packages/salesforcedx-sobjects-faux-generator/src/generator/typingGenerator.ts

```diff
@@ -20,33 +20,31 @@ export class TypingGenerator implements SObjectGenerator {
     this.declGenerator = new DeclarationGenerator();
   }
 
-  public generate(output: SObjectRefreshOutput): void {
+  public async generate(output: SObjectRefreshOutput): Promise<void> {
     const typingsFolderPath = path.join(output.sfdxPath, ...TYPING_PATH);
-    this.generateTypes([...output.getStandard(), ...output.getCustom()], typingsFolderPath);
+    await this.generateTypes([...output.getStandard(), ...output.getCustom()], typingsFolderPath);
   }
 
-  public generateTypes(sobjects: SObject[], targetFolder: string): void {
-    if (!fs.existsSync(targetFolder)) {
-      fs.mkdirSync(targetFolder, { recursive: true });
+  public async generateTypes(sobjects: SObject[], targetFolder: string): Promise<void> {
+    if (!(await folderExists(targetFolder))) {
+      await createDirectory(targetFolder);
     }
 
     for (const sobj of sobjects) {
       if (sobj.name) {
         const sobjDefinition = this.declGenerator.generateSObjectDefinition(sobj);
-        this.generateType(targetFolder, sobjDefinition);
+        await this.generateType(targetFolder, sobjDefinition);
       }
     }
   }
 
-  public generateType(folderPath: string, definition: SObjectDefinition): string {
+  public async generateType(folderPath: string, definition: SObjectDefinition): Promise<string> {
     const typingPath = path.join(folderPath, `${definition.name}${TYPESCRIPT_TYPE_EXT}`);
-    if (fs.existsSync(typingPath)) {
-      fs.unlinkSync(typingPath);
+    if (await folderExists(typingPath)) {
+      await deleteFile(typingPath);
     }
 
-    fs.writeFileSync(typingPath, this.convertDeclarations(definition), {
-      mode: 0o444
```

Comment:
was this important for some reason?  Is there another way to ensure that these are read-only?

Reply:
I tried workarounds but looks like there's no way to create read-only files using vscode api